apps/server/src/modules/tool/external-tool/controller/api-test/tool.api.spec.ts

```diff
@@ -723,4 +723,80 @@ describe('ToolController (API)', () => {
 			});
 		});
 	});
+
+	describe('[GET] tools/external-tools/:externalToolId/datasheet', () => {
+		describe('when user is not authenticated', () => {
+			const setup = () => {
+				const toolId: string = new ObjectId().toHexString();
+
+				return { toolId };
+			};
+
+			it('should return unauthorized', async () => {
+				const { toolId } = setup();
+
+				const response: Response = await testApiClient.get(`${toolId}/datasheet`);
+
+				expect(response.statusCode).toEqual(HttpStatus.UNAUTHORIZED);
+			});
+		});
+
+		describe('when externalToolId is given', () => {
+			const setup = async () => {
+				const toolId: string = new ObjectId().toHexString();
+				const externalToolEntity: ExternalToolEntity = externalToolEntityFactory.buildWithId(undefined, toolId);
+
+				const { adminUser, adminAccount } = UserAndAccountTestFactory.buildAdmin({}, [Permission.TOOL_ADMIN]);
+				await em.persistAndFlush([adminAccount, adminUser, externalToolEntity]);
+				em.clear();
+
+				const loggedInClient: TestApiClient = await testApiClient.login(adminAccount);
+
+				const date = new Date();
```

Comment:
This is problematic with timing, since the time this date is created is a different one than the one used in the uc to create the name.

Reply:
the date that is checked here is converted to a daily precision, so if the test isn't done at 23:59:59:990 or something, the will be no problem here..